src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx

```diff
@@ -1009,48 +1009,36 @@ XclBinUtilities::createMemoryBankGrouping(XclBin & xclbin)
   }
 }
 
-int transform_PDI_file(const std::string& fileName)
+#ifndef _WIN32
+// pdi_transform is only available on Linux
+// pdi_transform is defined in libtransformcdo.a
+extern "C" int pdi_transform(char* pdi_file, char* pdi_file_out);
+
+int transform_PDI_file(std::string fileName)
 {
-  // prototype: run the transform_static executable
-  // in the future, we may decide to link in the transform so instead
- 
-  // TODO: figure out whether we need to integrate transform_static into 
-  // XRT repo and how if so
-  // for now assume user has built or downloaded transform_static
-  // separately, it is either in the current working directory or in PATH 
-  std::string transformExePath = "./transform_static";
-  if (!fs::exists("./transform_static")) {
-#if (BOOST_VERSION >= 106400)
-    auto path = boost::process::search_path("transform_static");
-    if (!path.empty()) {
-      // std::cout << "Found executable at: " << path << std::endl;
-      transformExePath = path.string();
-    } else {
-      auto errMsg = boost::format("ERROR: --transform-pdi is specified, but transform_static executable is not found in the current working directory '%s' or PATH. Please make sure the exetuable is in PATH or CWD") % fs::current_path();
-      throw std::runtime_error(errMsg.str());
-    }
-#else
-    auto errMsg = boost::format("ERROR: --transform-pdi is specified, but transform_static executable is not found in the current working directory '%s'. Please copy the executable to CWD") % fs::current_path();
-    throw std::runtime_error(errMsg.str());
-#endif
+  // pdi_transform prints lots of messages
+  // redirect the output to a stream, so that console looks cleaner
+  int sout = dup(fileno(stdout));
+  FILE* file = freopen("/dev/null","w",stdout);
+  if (file == nullptr)
+    std::cout << "stdout redirect failed" << std::endl;
+
+  // pdi_transform can either return 0 or ‐1
+  int ret = pdi_transform(fileName.data(), fileName.data());
+  if (ret != 0) {
```

Comment:
We are using the same filename for both input and output?

Reply:
FILE* file here is only used to redirect the stdout so that output of pdi_transform() won't pollute the console
It has nothing to do with input